packages/devtools_app/lib/src/code_size/code_size_controller.dart

```diff
@@ -118,9 +118,18 @@ class CodeSizeController {
     changeDiffRoot(_activeDiffRoot);
   }
 
-  void loadTreeFromJsonFile(DevToolsJsonFile jsonFile) {
+  /// Notifies that the json files are currently being processed.
+  ValueListenable get processingNotifier => _processingNotifier;
+  final _processingNotifier = ValueNotifier<bool>(false);
+
+  Future<void> loadTreeFromJsonFile(DevToolsJsonFile jsonFile) async {
     changeSnapshotJsonFile(jsonFile);
 
+    _processingNotifier.value = true;
+
+    // Free up the thread for the code size page to display the loading message.
+    await delayForBatchProcessing(micros: 10);
```

Comment:
is this necessary? here and for the diff case below

Reply:
having micros as 0 didn't seem to be enough for the screen to display the loading text. Was surprised to see this given that the documentation states 0 micros should be enough.